src/Database/Redis/Cluster.hs

```diff
@@ -443,24 +443,24 @@ allMasterNodes (Connection nodeConns _ _ _ _) (ShardMap shardMap) =
 requestNode :: NodeConnection -> [[B.ByteString]] -> IO [Reply]
 requestNode (NodeConnection pool lastRecvRef nid) requests = do
     envTimeout <- round . (\x -> (x :: Time.NominalDiffTime) * 1000000) . realToFrac . fromMaybe (0.5 :: Double) . (>>= readMaybe) <$> lookupEnv "REDIS_REQUEST_NODE_TIMEOUT"
-    eresp <- race requestNodeImpl (threadDelay envTimeout)
+    eresp <- race (withResource pool requestNodeImpl) (threadDelay envTimeout)
     case eresp of
         Left e -> return e
         Right _ -> putStrLn ("timeout happened" ++ show nid) *> throwIO NoNodeException
```

Comment:
@shashitnak , lets create a custom exception for Timeout, we can name it RedisTimeoutException or something.
There might be other reasons for timeout apart from NoNode only

Reply:
@ishan-juspay I actually thought about this. There is a test where an infinite script is sent to a node which causes it to timeout, which we aren't handling currently. So apart from NoNodeException, I'm thinking about adding a BusyNodeException

Reply:
@ishan-juspay added a RedisTimeoutException type but it's still throwing NoNodeException wrapped in RedisTimeoutException for now. Will add the logic for figuring out the appropriate Exception later